nir/src/main/scala/scala/scalanative/nir/Fresh.scala

```diff
@@ -3,13 +3,31 @@ package nir
 
 import java.util.concurrent.atomic.AtomicInteger
 
-final class Fresh(scope: String) {
-  private var i = new AtomicInteger(0)
-  def apply() = {
-    val value = i.getAndIncrement()
-    Local(scope, value)
+final class Fresh private (private var start: Int) {
+  def apply(): Local = {
+    start += 1
+    val value = start
+    Local(value)
   }
 }
+
 object Fresh {
-  def apply(scope: String) = new Fresh(scope)
+  def apply(start: Int = 0): Fresh =
+    new Fresh(start)
+
+  def apply(insts: Seq[Inst]): Fresh = {
+    var max = -1
+    insts.foreach {
+      case Inst.Let(local, _) =>
+        max = Math.max(max, local.id)
+      case Inst.Label(local, params) =>
+        max = Math.max(max, local.id)
+        params.foreach { param =>
+          max = Math.max(max, param.name.id)
+        }
+      case _ =>
+        ()
+    }
+    new Fresh(max)
+  }
```

Comment:
My understanding is that:
 - We do this to be able to deterministically generate an instance of `Fresh` for a given  `Define`.
 - Because things are run in a non-deterministic order, we can't easily give a unique, deterministic number to each instance of `Fresh`.
 - This introduces a full pass over all the instructions of all `Define`s at the beginning of every pass.

If I've understood that correctly, couldn't we derive a unique, deterministic prefix from the `Define`? Or is the cost of this traversal negligible?